docs/src/evaluating.md

```diff
@@ -0,0 +1,372 @@
+---
+id: evaluating
+title: "Evaluating JavaScript"
+---
+
+Playwright scripts run in your Playwright environment. Your page scripts run in the browser page environment. Those environments don't intersect, they are running in different virtual machines in different processes and even potentially on different computers.
+
+The [`method: Page.evaluate`] API can run a JavaScript function in the context
+of the web page and bring results back to the Playwright environment. Browser globals like
+`window` and `document` can be used in `evaluate`.
+
+```js
+const href = await page.evaluate(() => document.location.href);
+```
+
+```java
+String href = (String) page.evaluate("document.location.href");
```

Comment:
Why string and not a function like other languages?

Reply:
I don't know!